src/index.ts

```diff
@@ -32,6 +32,21 @@ export function genericAuthMiddleware(fn: () => Promise<string | null | undefine
   })
 }
 
+export function locationMiddleware(
+  fn: () => Promise<string | null | undefined>
+) {
+  return setContext(async (_, context) => {
+    const key = await fn();
+
+    const headers = {
+      ...context.headers,
+      'X-Kanvas-Location': key ? key : '',
```

Comment:
@RivierGrullon dont send it empty , its better not to send it if you dont have any data